dirsrvtests/tests/suites/webui/ldap_browser/ldap_browser_test.py

```diff
@@ -70,6 +70,284 @@ def test_ldap_browser_tab_visibility(topology_st, page, browser_name):
+         8. Tree view is reset.
+         9. Pop-up window will appear with creation details.
+         10. Success
+         11. Testou is created.
+         12. Pop-up window will appear with creation details.
+         13. Success
+         14. Testrole is created
+         15. Pop-up window will appear with creation details.
+         16. Success
+         17. Custom entry with uid=1234 is created.
+    """
+    setup_login(page)
+    time.sleep(1)
+    frame = check_frame_assignment(page, browser_name)
+
+    log.info('Click on LDAP Browser tab, expand dc=example,dc=com tree and click on ou=people.')
+    frame.get_by_role('tab', name='LDAP Browser', exact=True).click()
+    frame.get_by_role('button').filter(has_text='dc=example,dc=com').click()
+    frame.get_by_role('button').filter(has_text='ou=people').click()
+    frame.get_by_role('columnheader', name='Attribute').wait_for()
+    time.sleep(1)
+
+    log.info('Create a new user named John Smith.')
+    frame.get_by_role("tabpanel", name="Tree View").get_by_role("button", name="Actions").click()
+    frame.get_by_role("menuitem", name="New ...").click()
+    frame.get_by_role("button", name="Next").click()
+    frame.get_by_role("button", name="Options menu").click()
+    frame.get_by_role("option", name="Posix Account").click()
+    frame.get_by_role("button", name="Next", exact=True).click()
+    frame.get_by_role("button", name="Next", exact=True).click()
+    frame.get_by_role("button", name="Place row 0 in edit mode").click()
+    frame.get_by_role("textbox", name="_").fill("John Smith")
+    frame.get_by_role("button", name="Save row edits for row 0").click()
+    frame.get_by_role("button", name="Place row 1 in edit mode").click()
+    frame.get_by_role("textbox", name="_").fill("John Smith")
+    frame.get_by_role("button", name="Save row edits for row 1").click()
+    frame.get_by_role("button", name="Place row 2 in edit mode").click()
+    frame.get_by_role("textbox", name="_").fill("1204")
+    frame.get_by_role("button", name="Save row edits for row 2").click()
+    frame.get_by_role("button", name="Place row 3 in edit mode").click()
+    frame.get_by_role("textbox", name="_").fill("/user/jsmith")
+    frame.get_by_role("button", name="Save row edits for row 3").click()
+    frame.get_by_role("button", name="Place row 4 in edit mode").click()
+    frame.get_by_role("textbox", name="_").fill("1204")
+    frame.get_by_role("button", name="Save row edits for row 4").click()
+    frame.get_by_role("button", name="Place row 5 in edit mode").click()
+    frame.get_by_role("textbox", name="_").fill("1204")
+    frame.get_by_role("button", name="Save row edits for row 5").click()
+    frame.get_by_role("button", name="Next").click()
+    frame.get_by_role("contentinfo").get_by_role("button", name="Create User").click()
+    frame.get_by_role("button", name="Finish").click()
+    frame.get_by_role("button").filter(has_text="cn=John Smith").wait_for()
```

Comment:
I think a lot of the code can be encapsulated into something like `create_entry(frame, entry_type, entry_data)` function where entry_data would be a dict with rows and values.

And something similar for another test function.